### ISSUE TITLE
attributes-order doesn't specify where slot-scope goes.

### ISSUE THREAD

**Issue by GufNZ** [vuejs/eslint-plugin-vue#432]
**Tell us about your environment**

* **ESLint Version:** 4.18.2
* **eslint-plugin-vue Version:** 4.3.0
* **Node Version:** 8.9.4

**Please show your full configuration:** N/A

**What did you do? Please include the actual source code causing the issue.**
I have a template that uses a component supporting scoped slots, but the rule `vue/attributes-order` does not specify where the `slot-scope` attribute should go and treats it like a normal attribute in terms of order.

Before I updated to the version including this rule, I'd always written them first, like below:

```vue
<template>
   <component-with-scoped-slots>
      <div slot="left" slot-scope="{ someVar }" :class="leftSlotClasses">...</div>
      <div slot="right" slot-scope="{ someVar }" :class="rightSlotClasses">...</div>
   </component-with-scoped-slots>
</template>
```

**What did you expect to happen?**
I expected the position of `slot` and `slot-scope` to be the same in this rule.

**What actually happened? Please include the actual, raw output from ESLint.**
Since `slot-scope` is not explicitly covered, it is treated like any other custom attr, and expected to be after any bound ones.
`[eslint] Attribute ":class" should go before "slot-scope". (vue/attributes-order)`

~I can make them read at the expected positions thusly:~
```vue
   <div slot="left" :slot-scope="'{ someVar }'" :class="leftSlotClasses">...</div>
```
~but that feels like cheating..~

**Comment by GufNZ** [comment]
Related to this, I'd have expected the `name` attribute for `<slot>` tags to also be near the start, but again, it's treated like a normal attribute.

**Comment by GufNZ** [comment]
Should something be done for `<slot name=>`, I wonder if it could be generalised to work for `<input type= name=>` as well?  Just thinking out loud here..

**Comment by GufNZ** [comment]
Just realised my 'fix' of binding `slot-scope` breaks things entirely.  It is ignored, and references to the vars that should have been defined result in errors at render time.

**Comment by GufNZ** [comment]
Just noticed the tests for this rule don't cover `slot` anyway.
Will have to sort those outside of work hours..

**Comment by michalsnik** [comment]
Thanks for posting this issue and working on it @GufNZ! Let me know once it's ready so I can review :) Let's leave `<slot>` case for now and focus on treating `slot-scope` as equal in terms of order to `slot` attribute.

**Comment by GufNZ** [comment]
@michalsnik  The PR fixes just that, and I was going to make sure the tests cover the new change, but they don't yet cover `slot` anyway, so it could go in as is I guess.  All the change does is add `slot-scope` to the same category as `slot`, namely `UNIQUE`.
